src/code/qualification/plotQualificationPKRatio.m

```diff
@@ -43,93 +32,190 @@
             break
         end
     end
-    Study(i)=table2array(ObservedData(ObservedData.ID==PKRatio.ObservedDataRecordId,'Study'));
+    if isempty(ObservedData.ID(ObservedData.ID==PKRatio.ObservedDataRecordId))
+        ME = MException('plotQualificationPKRatio:notFoundInPath', ...
+            'In PK Ratio plot %d, Ratio %d, Study ID "%d" was not found in Observed Dataset', figureHandle, i, PKRatio.ObservedDataRecordId);
+        throw(ME);
+    end
+    
+    Study = table2array(ObservedData(ObservedData.ID==PKRatio.ObservedDataRecordId,'Study'));
+    Result.Study{i,1} = Study{1};
     
     % Load the mapped Time Profile Simulation Results
-    [csvSimFile, xmlfile] = getSimFile(PKRatio, SimulationMappings);
+    [csvSimFile, xmlfile] = getSimFile(PKRatio, SimulationMappings, REInputPath);
+    if isempty(csvSimFile)
+        ME = MException('plotQualificationPKRatio:notFoundInPath', ...
+            'In PK Ratio plot %d, Ratio %d, Project "%s" or Simulation "%s" was not found in SimulationMappings', figureHandle, i, PKRatio.Project, PKRatio.Simulation);
+        throw(ME);
+    end
     SimResult = loadSimResultcsv(csvSimFile, PKRatio.Simulation);
     
-    initSimulation(xmlfile,'none');
-    drugmass = getParameter('*Application_*|ProtocolSchemaItem|DrugMass',1,'parametertype','readonly');
-    Compound{i}=PKRatio.Project;
-    AGE(i) = getParameter('*|Organism|Age',1,'parametertype','readonly');
-    BW(i) = getParameter('*|Organism|Weight',1,'parametertype','readonly');
-    MW(i) = getParameter(sprintf('*|%s|Molecular weight',Compound{i}),1,'parametertype','readonly');
+    % All the output are kept so far, may be removed if not necessary
+    [AGE, BW, MW, drugmass] = getInfofromSimulation(xmlfile, PKRatio.Output);
     
-    X(i) = getParameter(sprintf('*|Organism|%s', xAxesOptions.Dimension),1,'parametertype','readonly');
+    Result.AGE(i, 1)=AGE;
+    Result.BW(i, 1)=BW;
+    Result.MW(i, 1)=MW;
+    Result.drugmass(i, 1)=drugmass(1);
     
-    % Use the right Output
+    % Get the right PK Output
     for j=1:length(SimResult.outputPathList)
         findPathOutput = strfind(SimResult.outputPathList{j}, PKRatio.Output);
         if ~isempty(findPathOutput)
-            predicted=SimResult.y{j};
-            predictedTime=SimResult.time;
-            predictedTimeUnit=SimResult.timeUnit;
+            Pred=SimResult.y{j};
+            PredUnit=SimResult.outputUnit{j};
+            SimTime=SimResult.time;
+            SimTimeUnit=SimResult.timeUnit;
+            
+            PredDimension = findDimensionfromUnit(PredUnit);
             break
         end
     end
+    if isempty(findPathOutput)

```

Comment:
@pchelle same here: do you check somewhere that `SimResult.outputPathList `is not empty?